dotnet/src/Experimental/Orchestration.Flow.IntegrationTests/CollectEmailPlugin.cs

```diff
@@ -47,29 +46,33 @@ public CollectEmailPlugin(Kernel kernel)
     [Description("Useful to assist in configuration of email address, must be called after email provided")]
     public async Task<string> CollectEmailAsync(
         [Description("The email address provided by the user, pass no matter what the value is")]
+        // ReSharper disable once InconsistentNaming
+#pragma warning disable CA1707 // Identifiers should not contain underscores
         string email_address,
-        ContextVariables variables)
+#pragma warning restore CA1707 // Identifiers should not contain underscores
+        KernelArguments arguments)
     {
-        var chat = this._chat.CreateNewChat(SystemPrompt);
+        var chat = new ChatHistory(SystemPrompt);
         chat.AddUserMessage(Goal);
 
-        ChatHistory? chatHistory = variables.GetChatHistory();
-        if (chatHistory?.Any() ?? false)
+        ChatHistory? chatHistory = arguments.GetChatHistory();
+        if (chatHistory?.Count > 0)
         {
             chat.AddRange(chatHistory);
         }
 
         if (!string.IsNullOrEmpty(email_address) && IsValidEmail(email_address))
         {
-            variables["email_address"] = email_address;
-
             return "Thanks for providing the info, the following email would be used in subsequent steps: " + email_address;
         }
 
         // invalid email, prompt user to provide a valid email
-        variables["email_address"] = string.Empty;
-        variables.PromptInput();
-        return await this._chat.GenerateMessageAsync(chat, this._chatRequestSettings).ConfigureAwait(false);
+        arguments["email_address"] = string.Empty;
+        arguments.PromptInput();
```

Comment:
Why are the arguments are added/modified even though they are not used by the code below? Is it to be used by the orchestrator or a subsequent function? If that is the case, consider returning them from the function and passing them as arguments to the subsequent function on a higher layer, managed by the orchestrator.

Reply:
in the context of flow orchestrator, both orchestrator and plugins would try to provide the required variables.
in this particular case, orchestrator configured the variable however plugin did additional validation, and decided to invalidate the value by clearing it.

It is a bit anti-pattern here.
in next phase, plan to make the plugin development more intuitive and code-free.